sympy/solvers/tests/test_solveset.py

```diff
@@ -183,6 +183,19 @@ def test_invert_complex():
     assert invert_complex(sinh(x), 0, x) != (x, FiniteSet(0))
 
 
+def test_invert_modular():
+    # issue #13178
+    n = symbols('n', integer=True)
+    a = 742938285
+    z = 1898888478
+    m = 2**31 - 1
+    x = 20170816
+    f = x - Mod(a**n*z, m)
+    assert solveset(f, n, S.Integers) == FiniteSet(100)
+    x = symbols('x')
+    assert solveset(2 - Mod(4*x, 5), x, S.Integers) == FiniteSet(3)
```

Comment:
3 is just one of infinitely many solutions (3, 8, 13, ...). The idea of solveset is to return the set of all solutions, not just one of them.

Reply:
I think this could be done to resolve this thing:
```
try:
            lhs, rhs_s = inverter(f, 0, symbol)
        except TypeError:
            lhs, rhs_s = inverter(f, 0, m, symbol)
            n = Dummy('n', integer=True)
            if not sym_in_expo and lhs == symbol:
                ans = list(rhs_s)[0]
                rhs_s = ImageSet(Lambda(n, m*n + ans), S.Integers)
```
where `sym_in_expo` is a boolean variable which determines wether the` symbol` is in exponent of the first argument of `Mod`